File: C/Algorithms/1534.c
int countGoodTriplets(int* arr, int arrSize, int a, int b, int c){
    int ai, aj, ak, abs, abstemp, count = 0;
    // Build combinations:
    for(unsigned i = 0; i < arrSize; i++){
        ai = arr[i];
        for(unsigned j = i + 1; j < arrSize; j++){
            aj = arr[j];
            // absolute val.
            abstemp = ai - aj;
            abs = abstemp < 0 ? -abstemp : abstemp;
            // Bail early if we don't satisfy first condition.
            if (a >= abs){
                for(unsigned k = j+1; k < arrSize; k++){
                    ak = arr[k];
                    // absolute val.
                    abstemp = aj - ak;
                    abs = abstemp < 0 ? -abstemp : abstemp;
                    // Check the rest of the conditions.
                    if (b >= abs){
                        abstemp = ai - ak;
                        abs = abstemp < 0 ? -abstemp : abstemp;
                        if (c >= abs)
                            count++;
                    }        
                }
            }
        }
    }
    return count;
}

File: Javascript/Algorithms/1534.js
/**
 * @param {number[]} arr
 * @param {number} a
 * @param {number} b
 * @param {number} c
 * @return {number}
 */
var countGoodTriplets = function(arr, a, b, c) {
    let ai, aj, ak, count = 0;
    const len = arr.length;
    const abs = Math.abs;
    // Go through combinations.
    for(let i = 0; i < len; i++){
        ai = arr[i];
        for(let j = i+1; j < len; j++){
            aj = arr[j];
            // Bail early if first condition isn't met.
            if (a >= abs(ai - aj)){
                for (let k = j+1; k < len; k++){
                    ak = arr[k];
                    // check the rest of the conditions.
                    if ((b >= abs(aj - ak)) && (c >= abs(ai - ak))){
                        count += 1;
                    }
                }
            }
        }
    }
    return count;
};
